Psyphoza - Mushroom Species & Blindness Improvments

DrDuckedGoose opened the pull request:
## About The Pull Request
Adds Psyphoza, a psychic blind species themed around mushrooms. 
Additionally, adds improvements for the blind trait that goes with this species.

<details>
<summary>Psyphoza</summary>

#### Characteristics
* **Customization** - Psyphoza are themed around mushrooms and allow players to choose their mushroom-cap and its respective color. 
* * **Naming** - Psyphoza names are constructed as following ``[mushroom genus] the [typically high number]``, such as ``Ganoderma the 656th``
* **Diet** -  Psyphoza can eat almost anything, except dairy. This includes a preference for gross & raw foods. However, milk will not sit right with them.
*  **Language** - Psyphoza know several languages, including Galactic Common, Sylvan(Plant people) and Sonus. Sonus is a species specific language for Psyphoza.
* **Damage Weaknesses** - Psyphoza are weak to fire, taking ``1.25x`` more fire damage than humans.
* **Blood** - Psyphoza blood is comprised of mushroom hallucinogens, consuming any type of drug, while low on blood, will restore the user's blood volume.

#### Abilities
* Psychic pings to vaguely sense their surroundings, extending to nearby rooms like a ghetto x-ray. However, information gathered from this ability is very basic and doesn't offer anything short of silhouettes. This allows them to effectively survey their surroundings and perform minor investigations. 
* Collaborative Spore Deposits. Psyphoza can create a deposit, or upgrade one, every two minutes. These deposits contain a message and heal Psyphoza standing near them, on it. Deposits generate resources, to a cap of [20 * upgrade amount], and can spend these to heal, at a rate of [3 * upgrade amount]. Deposits can only be upgraded to a tier of 3. 

<details>
<summary>Lore 🤮</summary>

🦐 

</details>

</details>
<details>
<summary>Blind Adjustments</summary>

* Moves overhead chat above the blind plane. Blind players can now see overhead text.
* Adds a 'ping' system that highlights the silhouette of talking mobs, the source of sounds and footsteps.
* Blind people can now use most tools they previously couldn't that were integral to performing their job, such as med-scanners.

</details>

## Why It's Good For The Game

Adds a new species that offers an intensely unique playstyle that functions within the current core gameplay loop. Complements a unique playstyle and offers unique lore to the setting.

Makes playing blind characters a smoother experience that can nurture genuine scenarios that aren't held down by looking at a black overlay. 

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

https://imgur.com/a/h3UORdb

![image](https://user-images.githubusercontent.com/40559528/206888993-565bebf2-8f4f-4b73-9797-cbf764ae7ab9.png)
![image](https://user-images.githubusercontent.com/40559528/206889018-b4c12fc2-42d2-43ba-87b2-9f161af9dd08.png)
![image](https://user-images.githubusercontent.com/40559528/206888999-d31fe32e-5ef8-4e45-b6ad-c5f7139d2b8c.png)
![image](https://user-images.githubusercontent.com/40559528/206889030-fb0586e9-f673-4bde-bfa3-ab313858e6ec.png)
![image](https://user-images.githubusercontent.com/40559528/206889035-a343a5f9-03c5-42b0-b2a1-be8bf0a7ec87.png)
![image](https://user-images.githubusercontent.com/40559528/206889040-00d1c5fb-c90e-48a3-9952-b8512eaff764.png)
![image](https://user-images.githubusercontent.com/40559528/206889047-fd47725e-479c-4def-a7d6-1a61bf58053e.png)
![image](https://user-images.githubusercontent.com/40559528/206889063-6179efb4-85b8-49b3-9cc0-b563fea2b487.png)
![image](https://user-images.githubusercontent.com/40559528/206889082-0621488f-85cb-448b-b54b-b6db857f8a4a.png)
![image](https://user-images.githubusercontent.com/40559528/206889088-691e7db6-85be-4269-bbe2-fd85ac2665f3.png)
![image](https://user-images.githubusercontent.com/40559528/206889121-3deb8b24-e19f-44a4-a461-4f9b90202e07.png)
![image](https://user-images.githubusercontent.com/40559528/206889136-0e733512-3f5a-436c-869f-7e0cc433061c.png)
![image](https://user-images.githubusercontent.com/40559528/206896691-f62cb998-b4c5-44e7-9b5f-695ec3f0cf06.png)

</details>

## Changelog
:cl: kit_katz#6614(Sprites) Racc-Off#3845(Code)
add: Add blind sense (``blind_send.dm``), a component that allows blind players to see noises
add: Add Sonus language (``sonus.dm``, ``language_holder.dm``)
add: Add Psyphoza species (``psypohza.dm``, ``_DEFINES/mobs.dm``, ``human.dm``)
add: Add Psyphoza limbs, organs & accessories (``psyphoza_bodyparts.dm``,  ``global_lists.dm``, ``_HELPERS/mobs.dm``, ``flavor_misc.dm``, ``create_mob.dm``, ``preferences.dm``, ``character_save.dm``, ``meat.dm``, ``character_parser.dm``, ``tongue.dm``, ``eyes.dm``, ``species.dm``, ``sprite_accessories.dm``)
add: Add variant of telekinesis, weak telekinesis (``DNA.dm``, ``telekinesis.dm``, ``_machinery.dm``, ``scanners.dm``, ``objs.dm``, ``states.dm``, ``lighting.dm``) 
add: Adds Bohemian Rhapsody 
add: Add global signal for played sounds (``global_signals.dm``, ``sound.dm``)
add: Add dedicated plane for the 'psychic plane' (``layers.dm``, ``plane_master_controller.dm``) plane used for highlighting psyphoza pings
tweak: Tweak plane layer values to make room for psychic plane  (``layers.dm``)
tweak: Tweak runechat plane to a higher value, rendering above the blind plane / layer (``layers.dm``)
tweak: Tweak footsteps sounds to play from the respective mob, instead of tiles (``footstep.dm``)
tweak: Tweak species restrictions on helmets and select space suits (``helmet.dm``, ``miscellaneous.dm``)
tweak: Tweak blind accessibility for items like med-scanners (``shuttle_creator_console.dm``, ``scanners.dm``)
tweak: tweak blind overlay options / control (``status_procs.dm``)
tweak: Tweak some mob's starting sprite to interface with blind sense (``monkey.dm``, ``human.dm``)
/:cl:

Community Approval - May concern blind tweaks
![image](https://user-images.githubusercontent.com/40559528/206888853-8c26b11d-b16f-4e3f-b1f7-fe6e390112a3.png)
Headmin Approval (Real 100% no fake working 2023) - Does not concern blind tweaks
![image](https://user-images.githubusercontent.com/40559528/206895292-5ce9a0fb-e7b1-4955-a3c4-b657841fa3e9.png)

DrDuckedGoose committed:
initial - 3 12 22

DrDuckedGoose committed:
Psychic vision base - 3 12 22

DrDuckedGoose committed:
artsy stuff - 3 12 22

DrDuckedGoose committed:
Better for maintenance - 4 12 22

DrDuckedGoose committed:
Fucking areas - 4 12 22

DrDuckedGoose committed:
filters r us - 4 12 22

DrDuckedGoose committed:
Fix constant x-ray - 4 12 22

DrDuckedGoose committed:
timers / cooldowns - 4 12 22

DrDuckedGoose committed:
Cache limit - 4 12 22

DrDuckedGoose committed:
Audio features & removal of restrictions - 5 12 22

DrDuckedGoose committed:
blind improvements - 5 12 22

DrDuckedGoose committed:
blind directional features - 5 12 22

DrDuckedGoose committed:
psyphoza blind improvements - 5 12 22

DrDuckedGoose committed:
blind xray & icon QOL - 5 12 22

DrDuckedGoose committed:
fix psychic directions - 5 12 22

DrDuckedGoose committed:
S'more QOL - 5 12 22

DrDuckedGoose committed:
hard dels - 6 12 22

DrDuckedGoose committed:
Guchi icons - 6 12 22

DrDuckedGoose committed:
swap over to planes - 7 12 22

DrDuckedGoose committed:
Fix pixel shifting - 7 12 22

DrDuckedGoose committed:
fix blind icons & sense cut & rearrange files - 7 12 22

DrDuckedGoose committed:
obscure names again - 7 12 22

sweaty hands :(

DrDuckedGoose committed:
Species TK - 7 12 22

DrDuckedGoose committed:
Fix icon displacement - 7 12 22

DrDuckedGoose committed:
no  helmets - 7 12 22

DrDuckedGoose committed:
Custom species stuff - 7 12 22

DrDuckedGoose committed:
species organs - 8 12 22

DrDuckedGoose committed:
mass rebrand to fungus - 8 12 22

DrDuckedGoose committed:
Fix TK & Blind issues - 8 12 22

DrDuckedGoose committed:
Actually fix issues - 9 12 22

DrDuckedGoose committed:
Handle hardsuit issues - 9 12 22

DrDuckedGoose committed:
overlay improvements - 10 12 22

DrDuckedGoose committed:
Fix cap rendering - 11 12 22

DrDuckedGoose committed:
More layer adjustments - 11 12 22

DrDuckedGoose committed:
More secretive icons - 11 12 22

DrDuckedGoose committed:
Clean up blind_sense removal & addition - 12 12 22

DrDuckedGoose committed:
Linter - 12 12 22

DrDuckedGoose committed:
Update status_procs.dm

whoops

DrDuckedGoose committed:
Custom screams - 12 12 22

DrDuckedGoose committed:
Merge branch 'master' into jelly-people

DrDuckedGoose committed:
Update psyphoza sound scheme & fix error in emote code - 12 12 22 fix

DrDuckedGoose committed:
Better offsets - 12 12 22

DrDuckedGoose committed:
Sprite tweaks & cap - 12 12 22

DrDuckedGoose committed:
visual adjustments - 13 12 22

DrDuckedGoose committed:
Update config - 13 12 22

DrDuckedGoose committed:
Fuck - 13 12 22

DrDuckedGoose committed:
Update psyphoza_caps.dmi

DrDuckedGoose committed:
Random names - 17 122 22

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Fix blind icons & psychic layering - 18 12 22

DrDuckedGoose committed:
Fix naming, blacklisting & eye overlay - 18 12 22

DrDuckedGoose committed:
Update blind_sense.dm

DrDuckedGoose committed:
Fix  SOME sense dir updating - 18 12 22

All the ones that matter

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
SM & Eyes - 18 12 22

DrDuckedGoose committed:
Fix in-hand icons - 18 12 22

Kapu1178 commented on code/datums/components/blind_sense.dm:
```diff
@@ -0,0 +1,87 @@
+	var/type
+	if(isliving(speaker))
+		type = "mob"
+		if(isfile(message)) //soundfile exclusively from footstep signal
```
doesnt work on filepaths. only works on `file()` objects.

DrDuckedGoose replied:
It just works
![image](https://user-images.githubusercontent.com/40559528/208799610-204b6a98-0516-4479-9cfd-206b36a583c9.png)
(No really, it does)

DrDuckedGoose replied:
Isn't this right, or am I missing what you said?
![image](https://user-images.githubusercontent.com/40559528/208799823-9fc66590-fb0c-4c10-af60-5b70d16d2912.png)
![image](https://user-images.githubusercontent.com/40559528/208800698-8a869b2a-c906-4611-ac57-19c79cb18d14.png)

Kapu1178 replied:
Apparently it does. Coulda sworn I tried to do the same a few months ago. Oh well!

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into jelly-people

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into jelly-people

DrDuckedGoose committed:
I hope people don't abuse this - 5 1 23

Behave god damn you

DrDuckedGoose committed:
TEST COMMIT - 5 1 23

Commit for testing, removes species TK

DrDuckedGoose committed:
Balance & Flavour - 11 1 23

PowerfulBacon commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,276 @@
+	M.plane = PSYCHIC_PLANE //Draw overlay on this plane so we can use it as a mask
+	M.dir = target.dir
+	//Add overlay for highlighting
+	target.add_overlay(M)
+	overlays += list(target, M)
```
Rather than adding an overlay, set the images loc var and add it to the client images variable (Could use override too).

DrDuckedGoose replied:
This causes issues with in-hand icons not being visible. I originally used the method you're describing then swapped off to fix the bug.

DrDuckedGoose replied:
Didn't reply to this after a couple days, I'm just going to mark this as resolved.

PowerfulBacon replied:
Unresolving since I believe add_overlay calls build_appearance which was having quite a lot of (add_overlay is usually called pretty infrequently, but this will be calling it in significant amounts).

PowerfulBacon replied:
hm, thats strange it breaks with in-hand icons if its using the appearance of another object. Was it that the apperance doesnt update when they take something out of their hand, or it just didnt show in the first place?
If its only in-hand icons it might have to be a case where we take the L and lose that in exchange for better performance

DrDuckedGoose replied:
The latter, it just didn't show in the first place. However, I don't think the last few tests have proven the performance to be particularly terrible.
Perhaps some more testing and we can figure out if it's fine as-is.

PowerfulBacon replied:
Alright, I'll have to log on during a testmerge but it might be a couple of weeks due to exams

DrDuckedGoose committed:
Fixes & QOL - 12 1 23

DrDuckedGoose committed:
Features & Silicon languages - 12 1 23

Dejaku51 commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -55,6 +55,12 @@
 		if(findname(.))
 			. = .(gender, TRUE, null, ++attempts)
 
+/datum/species/apid/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
```
wrong species

DrDuckedGoose replied:
You can tell where I yoinked code from, lol. Thanks mate.

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Collaborative spores - 12 1 23

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
linter

Archanial commented on code/__DEFINES/layers.dm:
```diff
@@ -153,15 +154,15 @@
 
-#define HUD_PLANE 23
-#define HUD_LAYER 23
+#define HUD_PLANE 24
```
Runechat shouldn't be on hud layer.
If you are moving stuff up one layer be consistent.

Archanial commented on code/datums/components/blind_sense.dm:
```diff
@@ -0,0 +1,87 @@
+	///Type casted to access client, owner
+	var/mob/owner
+	///What texture we use
+	var/texture = "texture_2"
```
this could use a better name

Archanial commented on code/_onclick/hud/plane_master_controller.dm:
```diff
@@ -87,4 +87,4 @@ INITIALIZE_IMMEDIATE(/atom/movable/plane_master_controller)
 /atom/movable/plane_master_controller/game
 	name = PLANE_MASTERS_GAME
-	controlled_planes = list(FLOOR_PLANE, GAME_PLANE, LIGHTING_PLANE)
+	controlled_planes = list(FLOOR_PLANE, GAME_PLANE, LIGHTING_PLANE, PSYCHIC_PLANE)
```
Shouldn't it get its own plane controller? Why is psychic plane controller by master controller?

Archanial commented on code/_onclick/telekinesis.dm:
```diff
@@ -12,27 +12,32 @@
 
-/obj/proc/attack_tk_grab(mob/user)
+/obj/proc/attack_tk_grab(mob/user, is_weak = FALSE)
+	if(is_weak)
```
This should be very specific override on a child

DrDuckedGoose replied:
Meant to completely remove this

Archanial commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -129,7 +129,7 @@ GENE SCANNER
 
 // Used by the PDA medical scanner too
 /proc/healthscan(mob/user, mob/living/M, mode = 1, advanced = FALSE, to_chat = TRUE)
-	if(isliving(user) && (user.incapacitated() || user.eye_blind))
```
Please do not change this logic

DrDuckedGoose replied:
The new logic is ``if(isliving(user) && (user.incapacitated()))``
This allows blind people to use med scanners.

Archanial replied:
I guess we can cope that they have tts interface or something

Archanial commented on code/modules/food_and_drinks/food/snacks/meat.dm:
```diff
@@ -147,6 +147,14 @@
 
+/obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/psyphoza
+	icon_state = "psyphoza meat"
+	desc = "Psychically awaiting consumtpion, spooky."
```
```suggestion
	desc = "Psychically awaiting consumption, spooky."
```
typo

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,359 @@
+///Burn mod for our species - we're weak to fire
+#define PSYPHOZA_BURNMOD 1.25
+///Invisibility layer for shrooms
+#define SHROOM_DEPOSIT_INVISIBILITY 99
```
missing undef

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,359 @@
+			end = "th"
+	. = "[pick(GLOB.psyphoza_first_names)] the [rand(1, 10) * 100 + num][end]"
+	if(unique && attempts < 10)
+		if(findname(.))
```
These two check can be merged

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,359 @@
+	. = "[pick(GLOB.psyphoza_first_names)] the [rand(1, 10) * 100 + num][end]"
+	if(unique && attempts < 10)
+		if(findname(.))
+			. = .(gender, TRUE, null, ++attempts)
```
```suggestion
			return .(gender, TRUE, null, ++attempts)
```

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,359 @@
+/datum/action/item_action/organ_action/psychic_highlight/New(Target)
+	. = ..()
+	//Setup massive blacklist typecache of non-renderables. Smaller than whitelist
+	sense_blacklist = typecacheof(list(/turf/open, /obj/machinery/door, /obj/machinery/light, /obj/machinery/firealarm,
```
Can this be made static?

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,359 @@
+	to_chat(user, "<span class='notice'>Available resources: [resources] \nThe deposit echos: '[message]'</span>")
+
+/obj/effect/psyphoza_spores/process(delta_time)
+	if(world.time % 3 == 0)
```
You should work around delta_time here.
You do not have a guarantee that this will be fired every x time, so you might miss some ticks when world.time would be divisible by 3.

Archanial commented on code/modules/shuttle/shuttle_creation/shuttle_creator_console.dm:
```diff
@@ -12,7 +12,7 @@
 	var/datum/action/innate/shuttle_creator/modify/modify_action = new
 
 /obj/machinery/computer/camera_advanced/shuttle_creator/check_eye(mob/user)
-	if(user.eye_blind || user.incapacitated())
```
I don't think this logic should be changed

DrDuckedGoose replied:
Only removed blind check

Archanial commented on code/modules/tgui/states.dm:
```diff
@@ -132,6 +132,6 @@
 
 /mob/living/carbon/shared_living_ui_distance(atom/movable/src_object, viewcheck = TRUE)
-	if(has_dna() && dna.check_mutation(TK) && tkMaxRangeCheck(src, src_object))
+	if(has_dna() && (dna.check_mutation(TK) || dna.check_mutation(TK_WEAK)) && tkMaxRangeCheck(src, src_object))
```
Can you make tk weak a child of tk? This would save you a lot of specific check mutations (probably)

Archanial commented on strings/names/psyphoza_first.txt:
```diff
@@ -0,0 +1,8 @@
+Russula
+Sarcodon
+Gomphus
+Leccinum
```
We could use some more random names here.
I suggest creating thread on forums or if you want, I can reach out to headmins so they can prepare naming guidelines and some names after this is approved.

Archanial reviewed (dismissed):
Very good code

DrDuckedGoose committed:
Review changes - 15 1 23

Co-Authored-By: Archanial <37456678+Archanial@users.noreply.github.com>

DrDuckedGoose committed:
Update code/modules/food_and_drinks/food/snacks/meat.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

DrDuckedGoose committed:
Review changes 2 - 15 1 23

Co-Authored-By: Archanial <37456678+Archanial@users.noreply.github.com>

DrDuckedGoose committed:
Merge branch 'jelly-people' of https://github.com/DrDuckedGoose/BeeStation-Hornet into jelly-people

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,362 @@
+///Invisibility layer for shrooms
+#define SHROOM_DEPOSIT_INVISIBILITY 99
+
+GLOBAL_LIST_INIT(psychic_sense_blacklist, list(/turf/open, /obj/machinery/door, /obj/machinery/light, /obj/machinery/firealarm,
```
```suggestion
GLOBAL_LIST_INIT(psychic_sense_blacklist, typecacheof(list(/turf/open, /obj/machinery/door, /obj/machinery/light, /obj/machinery/firealarm, {...}
```

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,362 @@
+	if(!eyes && istype(owner, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = owner
+		//eyes
+		eyes = eyes || locate(/obj/item/organ/eyes) in H.internal_organs
```
```suggestion
		eyes = locate(/obj/item/organ/eyes) in H.internal_organs
```
eyes is always null here

DrDuckedGoose replied:
``eyes`` is null at the start then assigned from internal organs. When those eyes are removed from the body ``eyes`` will be nulled. Otherwise ``eyes`` won't be null, having been assigned.  By checking, we can avoid calling ``locate`` every activation.

Archanial commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -0,0 +1,362 @@
+		eyes = eyes || locate(/obj/item/organ/eyes) in H.internal_organs
+		sight_flags = eyes?.sight_flags
+		//Register signal for losing our eyes
+		RegisterSignal(eyes, COMSIG_PARENT_QDELETING, .proc/handle_eyes)
```
Eyes can be null here

DrDuckedGoose committed:
Update psyphoza.dm

Co-Authored-By: Archanial <37456678+Archanial@users.noreply.github.com>

DrDuckedGoose committed:
Update psyphoza.dm

Co-Authored-By: Archanial <37456678+Archanial@users.noreply.github.com>

PestoVerde322 reviewed (commented):
![immagine](https://user-images.githubusercontent.com/75247747/212834771-6318607f-3ee1-46f0-a2ce-c378f3cb1d3d.png)
may need a better icon for this instead of a mushroom cap for their speech

Other than that icons seems, aside the lack of more caps variety imo

DrDuckedGoose committed:
Sprite improvements - 17 1 23

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into jelly-people

DrDuckedGoose committed:
Update layers.dm

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Fix planes - 20 1 23

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into jelly-people

DrDuckedGoose committed:
Merge branch 'master' into jelly-people

DrDuckedGoose committed:
Deposit colouring - 26 1 23

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
deposit improvements - 27 1 23

DrDuckedGoose committed:
Sprite adjustments - Sorry Pigeon - 27 1 23

feed them

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Fixes - 28 1 23

DrDuckedGoose committed:
feedback fixes - 29 1 23

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Merge branch 'master' into jelly-people

DrDuckedGoose committed:
Eep deebly - 9 2 23

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into jelly-people

DrDuckedGoose committed:
Fixes - 22 2 23

- Fixes seeing ghosts and other invisible objects
- Fix blind people slamming into walls
- Adds auto option to psychic sense
- Adds keybind for psychic sense
- Fix blind sense not removing
- Fix not being able to click blind sense icons

itsmeow commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -336,12 +353,58 @@ GLOBAL_LIST_INIT(psychic_sense_blacklist, typecacheof(list(/turf/open, /obj/mach
+/datum/action/change_psychic_auto/proc/parent_destroy()
+	SIGNAL_HANDLER
+
+	Destroy()
```
manually calling Destroy BAD
do qdel(src)

itsmeow commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -392,6 +455,26 @@ GLOBAL_LIST_INIT(psychic_sense_blacklist, typecacheof(list(/turf/open, /obj/mach
 	upgrades = min(upgrade_limit, upgrades+1)
 
+//Keybind for sense
+/datum/keybinding/mob/psychic_sense
```
put this in the keybind file please, and since the signal is generically named "species action primary" please make the name and description reflect this

itsmeow commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -392,6 +455,26 @@ GLOBAL_LIST_INIT(psychic_sense_blacklist, typecacheof(list(/turf/open, /obj/mach
+		return
+	if(!user.mob) 
+		return
+	PH = PH || locate(/datum/action/item_action/organ_action/psychic_highlight) in user.mob.actions
```
this should forward to their species datum and then be handled on the species

DrDuckedGoose replied:
https://www.youtube.com/watch?v=6AaKkGrTqjE

itsmeow commented on code/modules/mob/living/living.dm:
```diff
@@ -88,7 +88,7 @@
 		return
 	if(buckled || now_pushing)
 		return
-	if((confused || is_blind()) && stat == CONSCIOUS && (mobility_flags & MOBILITY_STAND) && m_intent == "run" && (!ismovable(A) || is_blocked_turf(A)) && !HAS_MOB_PROPERTY(src, PROP_CANTBUMPSLAM))  // ported from VORE, sue me
```
this is entirely the fault of the previous isblind changes. just make it check if you have the temporary kind of blindness (eyeblind) rather than eyeblind OR trait blindness. consider it a fix

itsmeow commented on code/modules/mob/living/status_procs.dm:
```diff
@@ -420,7 +420,7 @@
 	if(!is_blind())
 		adjust_blindness(-1)
-		var/datum/component/blind_sense/B = GetComponent(/datum/component/blind_sense)
+		var/datum/component/blind_sense/B = GetComponent(/datum/component/blind_sense)	
```
bad

itsmeow commented on code/modules/surgery/organs/eyes.dm:
```diff
@@ -417,7 +417,7 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	blind_type = /atom/movable/screen/fullscreen/blind/psychic
-	see_invisible = 99 //Magic number for shroom deposits, sorry
+	see_invisible = 55 //Magic number for shroom deposits, sorry
```
I will EXPLODE you. use a HUD

itsmeow replied:
No really, use a HUD

itsmeow commented on code/modules/surgery/organs/stomach.dm:
```diff
@@ -251,8 +251,9 @@
-		log_game("[key_name(owner)] as [owner] made a new spore deposit with the message [message] at [world.time]. [key_name(owner)] located at [AREACOORD(owner)]")
-		message_admins("[key_name(owner)] as [owner] made a new spore deposit with the message [message] at [world.time]. [key_name(owner)] located at [AREACOORD(owner)]")
+		log_game("[key_name(owner)] as [owner] made a new spore deposit with the message '[message]' at [world.time]. [key_name(owner)] located at [AREACOORD(owner)]")
+		message_admins("[key_name(owner)] as [owner] made a new spore deposit with the message '[message]' at [world.time]. [key_name(owner)] located at [AREACOORD(owner)]")
```
use `key_name_admin` (or set `include_links=TRUE`) for `message_admins`. do NOT give dchat access to ckeys. remove "as [owner]", that is redundant because key_name already includes their name. Do not include "at [world.time]" as log_game already includes this in its key. do not add a redundant key_name just to say their location, log_game already includes their location. it's fine in message_admins though.

Personally I found the admin messages really annoying and spammy, I would prefer no admin messages.

change log_game to `owner.log_message("", type=LOG_GAME)` to get the epic auto-key stuff

DrDuckedGoose committed:
Some review changes - 25 2 23

DrDuckedGoose committed:
OWNER DOESNT HAVE LOG GAME

Javlar

DrDuckedGoose committed:
Update stomach.dm

DrDuckedGoose committed:
species generic action - 27 2 23

itsmeow commented on code/modules/surgery/organs/stomach.dm:
```diff
@@ -251,7 +251,7 @@
 		if(!message)
 			return
-		owner.log_game("[key_name(owner)] as [owner] made a new spore deposit with the message '[message]'.", type=LOG_GAME)
+		log_game("[key_name(owner)] as [owner] made a new spore deposit with the message '[message]'.", type=LOG_GAME)
```
I did say log_message

itsmeow replied:
`owner.log_message("made a new spore deposit with the message '[message]'", type=LOG_GAME)`

itsmeow commented on code/modules/mob/living/carbon/human/species_types/psyphoza.dm:
```diff
@@ -52,6 +52,13 @@ GLOBAL_LIST_INIT(psychic_sense_blacklist, typecacheof(list(/turf/open, /obj/mach
+
+/datum/species/psyphoza/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	PH = locate(/datum/action/item_action/organ_action/psychic_highlight) in C.actions
```
this needs to be nulled on species loss

DrDuckedGoose committed:
review - 14 3 23

Co-Authored-By: itsmeow <10366817+itsmeow@users.noreply.github.com>

DrDuckedGoose committed:
Merge remote-tracking branch 'upstream/master' into jelly-people

DrDuckedGoose committed:
new standards - 14 3 23

Co-Authored-By: itsmeow <10366817+itsmeow@users.noreply.github.com>

DrDuckedGoose committed:
Update psyphoza.dm

DrDuckedGoose committed:
Update psyphoza.dm

itsmeow commented on code/modules/surgery/organs/stomach.dm:
```diff
@@ -215,3 +215,56 @@
+		var/message = input(owner, "What message would you like to imprint on the deposit?", "Deposit message:") as text|null
+		if(!message)
+			return
+		owner.log_message("made a new spore deposit with the message '[message]'", TRUE)
```
```suggestion
		owner.log_message("made a new spore deposit with the message '[message]'", type=LOG_GAME)
```

DrDuckedGoose committed:
Update stomach.dm

DrDuckedGoose committed:
Merge remote-tracking branch 'upstream/master' into jelly-people

DrDuckedGoose committed:
Merge branch 'master' into jelly-people

itsmeow reviewed (changes requested):
Mushroom deposits need to use a HUD rather than a snowflake byond invisibility
